src/backend/distributed/operations/shard_cleaner.c

```diff
@@ -727,17 +742,22 @@ TryLockRelationAndPlacementCleanup(Oid relationId, LOCKMODE lockmode)
  * true on success.
  */
 static bool
-TryDropShardOutsideTransaction(OperationId operationId, char *qualifiedTableName,
-							   char *nodeName, int nodePort)
+TryDropShardOutsideTransaction(OperationId operationId,
+							   char *qualifiedTableName,
+							   char *nodeName,
+							   int nodePort,
+							   bool logDropMessage)
 {
-	char *operation = (operationId == INVALID_OPERATION_ID) ? "move" : "cleanup";
-
-	ereport(LOG, (errmsg("cleaning up %s on %s:%d which was left "
-						 "after a %s",
-						 qualifiedTableName,
-						 nodeName,
-						 nodePort,
-						 operation)));
+	if (logDropMessage)
+	{
+		char *operation = (operationId == INVALID_OPERATION_ID) ? "move" : "cleanup";
+		ereport(LOG, (errmsg("cleaning up %s on %s:%d which was left "
+							 "after a %s",
+							 qualifiedTableName,
+							 nodeName,
+							 nodePort,
+							 operation)));
+	}
 
```

Comment:
shouldn't we be fixing this function for connection usage as well?

It may be useful to have a ExecuteCommandListInLocalSubtransaction that does the right thing.

Reply:
The changes will be part of other PR as discussed.

Reply:
Will do as part of other PR.